test/e2e/apimachinery/crd_conversion_webhook.go

```diff
@@ -0,0 +1,295 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apimachinery
+
+import (
+	"time"
+
+	apps "k8s.io/api/apps/v1"
+	"k8s.io/api/core/v1"
+	rbacv1beta1 "k8s.io/api/rbac/v1beta1"
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+	utilversion "k8s.io/kubernetes/pkg/util/version"
+	"k8s.io/kubernetes/test/e2e/framework"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	_ "github.com/stretchr/testify/assert"
+)
+
+const (
+	secretCRDName      = "sample-custom-resource-conversion-webhook-secret"
+	deploymentCRDName  = "sample-crd-conversion-webhook-deployment"
+	serviceCRDName     = "e2e-test-crd-conversion-webhook"
+	roleBindingCRDName = "crd-conversion-webhook-auth-reader"
+)
+
+// TODO(mehdy): this should be 1.12.0
+var serverCRDConversionWebhookVersion = utilversion.MustParseSemantic("v1.11.0")
+
+var _ = SIGDescribe("CustomResourceConversionWebhook [Feature:CustomResourceWebhookConversion]", func() {
+	var context *certContext
+	f := framework.NewDefaultFramework("crd-webhook")
+
+	var client clientset.Interface
+
+	BeforeEach(func() {
+		client = f.ClientSet
+
+		// Make sure the relevant provider supports admission webhook
+		framework.SkipUnlessServerVersionGTE(serverCRDConversionWebhookVersion, f.ClientSet.Discovery())
+		framework.SkipUnlessProviderIs("gce", "gke", "local")
+
+		By("Setting up server cert")
+		context = setupServerCert(f.Namespace.Name, serviceCRDName)
+		createAuthReaderRoleBindingForCRDConversion(f, f.Namespace.Name)
+
+		deployCustomResourceWebhookAndService(f, imageutils.GetE2EImage(imageutils.CRDConversionWebhook), context)
+	})
+
+	AfterEach(func() {
+		cleanCRDWebhookTest(client, f.Namespace.Name)
+	})
+
+	It("Should be able to convert from CRD v1 to CRD v2", func() {
+		apiVersions := []v1beta1.CustomResourceDefinitionVersion{
+			{
+				Name:    "v1",
+				Served:  true,
+				Storage: true,
+			},
+			{
+				Name:    "v2",
+				Served:  true,
+				Storage: false,
+			},
+		}
+		testcrd, err := framework.CreateMultiVersionTestCRD(f, "stable.example.com", apiVersions,
+			&v1beta1.WebhookClientConfig{
+				CABundle: context.signingCert,
+				Service: &v1beta1.ServiceReference{
+					Namespace: f.Namespace.Name,
+					Name:      serviceCRDName,
+					Path:      strPtr("/convert"),
+				}})
+		if err != nil {
+			return
+		}
+		defer testcrd.CleanUp()
+		testCustomResourceConversionWebhook(f, testcrd.Crd, testcrd.DynamicClients)
+	})
+
+})
+
+func cleanCRDWebhookTest(client clientset.Interface, namespaceName string) {
+	_ = client.CoreV1().Services(namespaceName).Delete(serviceCRDName, nil)
```

Comment:
isn't the namespace deleted anyway?

Reply:
I guess so though we do this for admission webhooks here: https://github.com/kubernetes/kubernetes/blob/733a381ec42ede15cd0499bc502a0cba6612f3ca/test/e2e/apimachinery/webhook.go#L1067

Does it worth investigating? I mean no harm removing them manually, right?